Get last commits of reef 7.1

barakda opened the pull request:
last commit update
ver update

gbregman committed:
Do not send QoS messages is we're still busy processing the previous ones.
Fixes #1104

Signed-off-by: Gil Bregman <gbregman@il.ibm.com>
(cherry picked from commit 2a4c23a5f7d9780223536181f539bf33b3c3e94f)

barakda committed:
nvmeof gateway ver update - 1.2.19

Signed-off-by: Barak Davidov <barakda@il.ibm.com>

barakda committed:
atom job add

Signed-off-by: Barak Davidov <barakda@il.ibm.com>

baum commented on .github/workflows/build-container.yml:
```diff
@@ -9,9 +9,9 @@ on:  # yamllint disable rule:truthy
   pull_request:
     branches:
-      - '*'
+      - reef_7.1
```
'*' is fine here, no need to change this

baum commented on .github/workflows/build-container.yml:
```diff
@@ -522,7 +522,7 @@ jobs:
       matrix:
-        test: ["sanity", "state_transitions", "state_transitions_both_gws", "state_transitions_loop", "state_transitions_rand_loop", "late_registration", "late_registration_loop", "4gws", "4gws_loop", "namespaces", "namespaces_loop", "mtls"]
+        test: ["sanity", "state_transitions", "state_transitions_both_gws", "state_transitions_loop", "state_transitions_rand_loop", "late_registration", "late_registration_loop", "4gws", "4gws_loop", "namespaces", "namespaces_loop", "mtls", "set_qos", "set_qos_2ms"]
     runs-on: ubuntu-latest
```
set_qos fix was not backported to 7.1 yet, so we probably either add that fix to 7.1 or remove "set_qos", "set_qos_2ms" tests

baum commented on tests/ha/set_qos.sh:
```diff
@@ -0,0 +1,20 @@
+GW1_IP="$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' "$GW1_NAME")"
+cephnvmf="docker compose run --rm nvmeof-cli --server-address $GW1_IP --server-port 5500"
+
+$cephnvmf subsystem add --subsystem $NQN --max-namespaces 1024 --no-group-append
```
```suggestion
$cephnvmf subsystem add --subsystem $NQN --max-namespaces 1024
```

see [CI](https://github.com/ceph/ceph-nvmeof/actions/runs/13763443952/job/38484937555?pr=1153):
```
++ docker compose run --rm nvmeof-cli --server-address 192.168.13.3 --server-port 5500 subsystem add --subsystem nqn.2016-06.io.spdk:cnode1QOS --max-namespaces 1024 --no-group-append
error: unrecognized arguments: --no-group-append
usage: python3 -m control.cli [-h] [--format {text,json,yaml,plain,python}]
                              [--output {log,stdio}]
                              [--log-level {debug,DEBUG,info,INFO,warning,WARNING,error,ERROR,critical,CRITICAL}]
                              [--server-address SERVER_ADDRESS]
                              [--server-port SERVER_PORT]
                              [--client-key CLIENT_KEY]
                              [--client-cert CLIENT_CERT]
                              [--server-cert SERVER_CERT] [--verbose]
                              {version,gw,spdk_log_level,subsystem,listener,host,connection,namespace,ns,get_subsystems}
```

baum commented on .github/workflows/build-container.yml:
```diff
@@ -9,9 +9,9 @@ on:  # yamllint disable rule:truthy
+      - reef_7.1
   schedule:
-    - cron: '0 0 * * *'
+    - cron: '0 8 * * *'
```
In GitHub Actions, the schedule directive only works for workflows defined in the repository's default branch. If you define a scheduled workflow in a side branch (i.e., a not in `devel` branch), it won’t trigger automatically.

To schedule daily executions for a side branch, you can define the workflow in the `devel` branch and use the actions/checkout action with the ref parameter set to the target branch. This allows the workflow to run on the desired branch.

Suggestion:
```yaml

name: 7.1 Release Nightly Runs
on:
  schedule:
    - cron: '0 0 * * *' # Runs daily at midnight UTC
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          ref: 'reef-7.1' # Specify the release branch
      - run: echo "Running on reef-7.1"
      ...
 ```
This setup would ensure the workflow operates on the specified side branch while still being triggered from the default branch. 🚀

Commit:
Remove github actions which are no longer allowed

The error: Actions in this workflow must be: within a repository owned by ceph or created by GitHub.

- docker/setup-qemu-action
- andstor/file-existence-action
- docker/login-action

Signed-off-by: Alexander Indenbaum <aindenba@redhat.com>

barakda committed:
remove atom nightly

Signed-off-by: Barak Davidov <barakda@il.ibm.com>

baum reviewed (approved):
lgtm
